lib/api/route/typing.dart

```diff
@@ -0,0 +1,30 @@
+import '../core.dart';
+import '../model/events.dart';
+import 'messages.dart';
+
+
+/// https://zulip.com/api/set-typing-status
+Future<void> setTypingStatus(ApiConnection connection, {
+  required TypingOp op,
+  required MessageDestination destination,
+}) {
+  switch (destination) {
+    case StreamDestination():
+      final supportsTypeChannel = connection.zulipFeatureLevel! >= 248; // TODO(server-9)
+      final supportsStreamId = connection.zulipFeatureLevel! >= 215; // TODO(server-8)
+      return connection.post('setTypingStatus', (_) {}, 'typing', {
+        'type':  RawParameter((supportsTypeChannel) ? 'channel' : 'stream'),
+        'op':    RawParameter(op.toJson()),
+        'topic': RawParameter(destination.topic),
+        ...(supportsStreamId) ? {'stream_id': destination.streamId}
+                              : {'to': [destination.streamId]}
```

Comment:
A fun Dart 3 way to write this:
```suggestion
        if (supportsStreamId) 'stream_id': destination.streamId
        else                  'to': [destination.streamId],
```